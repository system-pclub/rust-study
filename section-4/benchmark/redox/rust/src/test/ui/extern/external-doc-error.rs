// normalize-stderr-test: "not-a-file.md:.*\(" -> "not-a-file.md: $$FILE_NOT_FOUND_MSG ("

#![feature(external_doc)]

#[doc(include = "not-a-file.md")]
pub struct SomeStruct; //~^ ERROR couldn't read

#[doc(include = "auxiliary/invalid-utf8.txt")]
pub struct InvalidUtf8; //~^ ERROR wasn't a utf-8 file

#[doc(include)]
pub struct MissingPath; //~^ ERROR expected path
                        //~| HELP provide a file path with `=`
                        //~| SUGGESTION include = "<path>"

#[doc(include("../README.md"))]
pub struct InvalidPathSyntax; //~^ ERROR expected path
                              //~| HELP provide a file path with `=`
                              //~| SUGGESTION include = "../README.md"

#[doc(include = 123)]
pub struct InvalidPathType; //~^ ERROR expected path
                            //~| HELP provide a file path with `=`
                            //~| SUGGESTION include = "<path>"

#[doc(include(123))]
pub struct InvalidPathSyntaxAndType; //~^ ERROR expected path
                                     //~| HELP provide a file path with `=`
                                     //~| SUGGESTION include = "<path>"

fn main() {}
