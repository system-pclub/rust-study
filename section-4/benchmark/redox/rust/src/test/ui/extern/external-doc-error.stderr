error: couldn't read $DIR/not-a-file.md: $FILE_NOT_FOUND_MSG (os error 2)
  --> $DIR/external-doc-error.rs:5:17
   |
LL | #[doc(include = "not-a-file.md")]
   |                 ^^^^^^^^^^^^^^^ couldn't read file

error: $DIR/auxiliary/invalid-utf8.txt wasn't a utf-8 file
  --> $DIR/external-doc-error.rs:8:17
   |
LL | #[doc(include = "auxiliary/invalid-utf8.txt")]
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ contains invalid utf-8

error: expected path to external documentation
  --> $DIR/external-doc-error.rs:11:7
   |
LL | #[doc(include)]
   |       ^^^^^^^ help: provide a file path with `=`: `include = "<path>"`

error: expected path to external documentation
  --> $DIR/external-doc-error.rs:16:7
   |
LL | #[doc(include("../README.md"))]
   |       ^^^^^^^^^^^^^^^^^^^^^^^ help: provide a file path with `=`: `include = "../README.md"`

error: expected path to external documentation
  --> $DIR/external-doc-error.rs:21:7
   |
LL | #[doc(include = 123)]
   |       ^^^^^^^^^^^^^ help: provide a file path with `=`: `include = "<path>"`

error: expected path to external documentation
  --> $DIR/external-doc-error.rs:26:7
   |
LL | #[doc(include(123))]
   |       ^^^^^^^^^^^^ help: provide a file path with `=`: `include = "<path>"`

error: aborting due to 6 previous errors

